src/scan.c

```diff
@@ -586,6 +588,56 @@ static int scan_module(struct context_data *ctx, int ep, int chain)
     return (time + m->time_factor * frame_count * base_time / bpm);
 }
 
+static void reset_scan_data(struct context_data *ctx)
+{
+	int i;
+	for (i = 0; i < XMP_MAX_MOD_LENGTH; i++) {
+		ctx->m.xxo_info[i].time = -1;
+	}
+	memset(ctx->p.sequence_control, 0xff, XMP_MAX_MOD_LENGTH);
+}
+
+#ifndef LIBXMP_CORE_PLAYER
+static void compare_vblank_scan(struct context_data *ctx)
+{
+	/* Calculate both CIA and VBlank time for certain long MODs
+	 * and pick the more likely (i.e. shorter) one. The same logic
+	 * works regardless of the initial mode selected--either way,
+	 * the wrong timing mode usually makes modules MUCH longer. */
+	struct player_data *p = &ctx->p;
+	struct module_data *m = &ctx->m;
+	struct ord_data *info_backup;
+	struct scan_data scan_backup;
+	unsigned char ctrl_backup[256];
+
+	if ((info_backup = malloc(sizeof(m->xxo_info))) != NULL) {
```

Comment:
A `(struct ord_data *) ` cast here

Reply:
OK, done. I'll combine any further tweaks into the new commit so this doesn't need to be squashed...